### PR TITLE
[CBRD-25257] Include headers of wrapping functions into "all" cubrid files

### PR DESCRIPTION
**Pull request by Rudeus**
http://jira.cubrid.org/browse/CBRD-25257

Include headers of wrapping functions into "all" .c / .cpp cubrid files with WARNING comment
 - Warning comment is about "memory_wrapper.hpp should be the last including header"
 - Except mvcc.c / btree_unique.cpp because they use `placement new` with no operator new for allocation
 - License check fail because of mallocheap.h

Add new workflows for checking about .c/.cpp files have `memory_wrapper.hpp`
 - Check memory_wrapper.hpp is included in .c/.cpp files
 - Check memory_wrapper.hpp is the last header that the file includes

### CODE REVIEW

File: src/thread/thread_daemon.cpp [hunk 1]
```diff
@@ -39,9 +40,9 @@ namespace cubthread
   static const cubperf::statset_definition Daemon_statistics =
   {
     cubperf::stat_definition (STAT_LOOP_EXECUTE_COUNT_AND_TIME, cubperf::stat_definition::COUNTER_AND_TIMER,
-			      "daemon_loop_count", "daemon_execute_time"),
+    "daemon_loop_count", "daemon_execute_time"),
```
Comment (Rudeus) [inline src/thread/thread_daemon.cpp]:
```suggestion
			      "daemon_loop_count", "daemon_execute_time"),
```

File: src/thread/thread_daemon.cpp [hunk 2]
```diff
@@ -39,9 +40,9 @@ namespace cubthread
...
     cubperf::stat_definition (STAT_LOOP_PAUSE_TIME, cubperf::stat_definition::TIMER,
-			      "daemon_pause_time")
+    "daemon_pause_time")
```
Comment (Rudeus) [inline src/thread/thread_daemon.cpp]:
```suggestion
			      "daemon_pause_time")
```

File: src/thread/thread_waiter.cpp [hunk 3]
```diff
@@ -43,10 +44,10 @@ namespace cubthread
     cubperf::stat_definition (STAT_LOCK_WAKEUP_COUNT, cubperf::stat_definition::COUNTER, "waiter_lock_wakeup_count"),
     cubperf::stat_definition (STAT_SLEEP_COUNT, cubperf::stat_definition::COUNTER, "waiter_sleep_count"),
     cubperf::stat_definition (STAT_TIMEOUT_COUNT, cubperf::stat_definition::COUNTER,
-			      "waiter_timeout_count"),
+    "waiter_timeout_count"),
```
Comment (Rudeus) [inline src/thread/thread_waiter.cpp]:
```suggestion
			      "waiter_timeout_count"),
```

File: src/thread/thread_waiter.cpp [hunk 4]
```diff
@@ -43,10 +44,10 @@ namespace cubthread
...
     cubperf::stat_definition (STAT_NO_SLEEP_COUNT, cubperf::stat_definition::COUNTER, "waiter_no_sleep_count"),
     cubperf::stat_definition (STAT_AWAKEN_COUNT_AND_TIME, cubperf::stat_definition::COUNTER_AND_TIMER,
-			      "waiter_awake_count", "waiter_wakeup_delay_time")
+    "waiter_awake_count", "waiter_wakeup_delay_time")
```
Comment (Rudeus) [inline src/thread/thread_waiter.cpp]:
```suggestion
			      "waiter_awake_count", "waiter_wakeup_delay_time")
```

File: src/thread/thread_worker_pool.cpp [hunk 5]
```diff
@@ -52,21 +53,21 @@ namespace cubthread
   static const cubperf::statset_definition Worker_pool_statdef =
   {
     cubperf::stat_definition (Wpstat_start_thread, cubperf::stat_definition::COUNTER_AND_TIMER,
-			      "Counter_start_thread", "Timer_start_thread"),
+    "Counter_start_thread", "Timer_start_thread"),
```
Comment (Rudeus) [inline src/thread/thread_worker_pool.cpp]:
```suggestion
			      "Counter_start_thread", "Timer_start_thread"),
```

File: src/thread/thread_worker_pool.cpp [hunk 6]
```diff
@@ -52,21 +53,21 @@ namespace cubthread
...
     cubperf::stat_definition (Wpstat_create_context, cubperf::stat_definition::COUNTER_AND_TIMER,
-			      "Counter_create_context", "Timer_create_context"),
+    "Counter_create_context", "Timer_create_context"),
```
Comment (Rudeus) [inline src/thread/thread_worker_pool.cpp]:
```suggestion
			      "Counter_create_context", "Timer_create_context"),
```

File: src/thread/thread_worker_pool.cpp [hunk 7]
```diff
@@ -52,21 +53,21 @@ namespace cubthread
...
     cubperf::stat_definition (Wpstat_execute_task, cubperf::stat_definition::COUNTER_AND_TIMER,
-			      "Counter_execute_task", "Timer_execute_task"),
+    "Counter_execute_task", "Timer_execute_task"),
```
Comment (Rudeus) [inline src/thread/thread_worker_pool.cpp]:
```suggestion
			      "Counter_execute_task", "Timer_execute_task"),
```

File: src/thread/thread_worker_pool.cpp [hunk 8]
```diff
@@ -52,21 +53,21 @@ namespace cubthread
...
     cubperf::stat_definition (Wpstat_retire_task, cubperf::stat_definition::COUNTER_AND_TIMER,
-			      "Counter_retire_task", "Timer_retire_task"),
+    "Counter_retire_task", "Timer_retire_task"),
```
Comment (Rudeus) [inline src/thread/thread_worker_pool.cpp]:
```suggestion
			      "Counter_retire_task", "Timer_retire_task"),
```

File: src/thread/thread_worker_pool.cpp [hunk 9]
```diff
@@ -52,21 +53,21 @@ namespace cubthread
...
     cubperf::stat_definition (Wpstat_found_in_queue, cubperf::stat_definition::COUNTER_AND_TIMER,
-			      "Counter_found_task_in_queue", "Timer_found_task_in_queue"),
+    "Counter_found_task_in_queue", "Timer_found_task_in_queue"),
```
Comment (Rudeus) [inline src/thread/thread_worker_pool.cpp]:
```suggestion
			      "Counter_found_task_in_queue", "Timer_found_task_in_queue"),
```

File: src/thread/thread_worker_pool.cpp [hunk 10]
```diff
@@ -52,21 +53,21 @@ namespace cubthread
...
     cubperf::stat_definition (Wpstat_wakeup_with_task, cubperf::stat_definition::COUNTER_AND_TIMER,
-			      "Counter_wakeup_with_task", "Timer_wakeup_with_task"),
+    "Counter_wakeup_with_task", "Timer_wakeup_with_task"),
```
Comment (Rudeus) [inline src/thread/thread_worker_pool.cpp]:
```suggestion
			      "Counter_wakeup_with_task", "Timer_wakeup_with_task"),
```

File: src/thread/thread_worker_pool.cpp [hunk 11]
```diff
@@ -52,21 +53,21 @@ namespace cubthread
...
     cubperf::stat_definition (Wpstat_recycle_context, cubperf::stat_definition::COUNTER_AND_TIMER,
-			      "Counter_recycle_context", "Timer_recycle_context"),
+    "Counter_recycle_context", "Timer_recycle_context"),
```
Comment (Rudeus) [inline src/thread/thread_worker_pool.cpp]:
```suggestion
			      "Counter_recycle_context", "Timer_recycle_context"),
```

File: src/thread/thread_worker_pool.cpp [hunk 12]
```diff
@@ -52,21 +53,21 @@ namespace cubthread
...
     cubperf::stat_definition (Wpstat_retire_context, cubperf::stat_definition::COUNTER_AND_TIMER,
-			      "Counter_retire_context", "Timer_retire_context")
+    "Counter_retire_context", "Timer_retire_context")
```
Comment (Rudeus) [inline src/thread/thread_worker_pool.cpp]:
```suggestion
			      "Counter_retire_context", "Timer_retire_context")
```

**Review by joohok — APPROVED** [review]

**Comment by hornetmj** [comment]
> Except mvcc.c / btree_unique.cpp because they use placement new with no operator new for allocation

How about replacing `placement new` as the following way:
- mvcc.c
```
void
mvcc_info::init ()
{
    snapshot = snapshot ();
    id = MVCCID_NULL;
    recent_snapshot_lowest_active_mvccid = MVCCID_NULL;
    sub_ids = sub_ids ();
}

or

one of the option is just to call mvcc_info::reset(). (need check)
```
- btree_unique.cpp - same approach

Additionally, we need to consider memory leak from existing `replacement new`.

File: src/heaplayers/heaps/top/mallocheap.h [hunk 13]
```diff
@@ -60,7 +60,7 @@ namespace HL
 	  void *p =::malloc (sz + cubmem::MMON_METAINFO_SIZE);
 	  if (p != NULL)
 	    {
-	      mmon_add_stat ((char *) p, sz + cubmem::MMON_METAINFO_SIZE, __FILE__, __LINE__);
```
Comment (hornetmj) [inline src/heaplayers/heaps/top/mallocheap.h]: This is already fixed in https://github.com/CUBRID/cubrid/pull/5188.

File: .github/workflows/check.yml [hunk 14]
```diff
@@ -195,3 +195,53 @@ jobs:
           msg: |
             **cppcheck**
             ${{ steps.get-comment-body.outputs.body }}
+  memory-header-check:
```
Comment (hornetmj) [inline .github/workflows/check.yml]: memory-monitor-check

File: src/heaplayers/heaps/top/mallocheap.h (hunk 13 above)
Comment (Rudeus) [reply src/heaplayers/heaps/top/mallocheap.h]: Done.

File: .github/workflows/check.yml (hunk 14 above)
Comment (Rudeus) [reply .github/workflows/check.yml]: Done.

**Review by hornetmj — APPROVED** [review]

**Pull request merged**